cadsapi deprecation

All build 3.X tests now fail:

```
==================================== ERRORS ====================================
_____________________ ERROR collecting tests/test_data.py ______________________
ImportError while importing test module '/home/runner/work/amundsen-sea-low-index/amundsen-sea-low-index/tests/test_data.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.[9](https://github.com/davidwilby/amundsen-sea-low-index/actions/runs/12084734625/job/33702610870#step:5:10).20/x64/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_data.py:8: in <module>
    import cads_api_client
E   ModuleNotFoundError: No module named 'cads_api_client'
=========================== short test summary info ============================
ERROR tests/test_data.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.[16](https://github.com/davidwilby/amundsen-sea-low-index/actions/runs/12084734625/job/33702610870#step:5:17)s ===============================
Error: Process completed with exit code 2.
```

[cads_api_client](https://github.com/ecmwf-projects/cads-api-client) has been archived and deprecated in favour of [datapi](https://github.com/ecmwf-projects/datapi).

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -18,7 +18,7 @@ classifiers = [
 ]
 requires-python = ">=3.9"
 dependencies = [
-    "cdsapi>=0.7.0",
+    "cdsapi>=0.7.5",
     "cartopy",
     "dask",
     "joblib",

--- a/tests/test_data.py
+++ b/tests/test_data.py
@@ -5,8 +5,8 @@ import unittest
 from unittest.mock import patch
 import uuid
 
-import cads_api_client
-import cads_api_client.legacy_api_client
+import datapi
+import datapi.legacy_api_client
 import pytest
 
 from asli import ASL_REGION
@@ -105,7 +105,7 @@ class TestCDSDownloader(unittest.TestCase):
             }
         )
 
-        with patch.object(cads_api_client.legacy_api_client.LegacyApiClient, "retrieve", return_value=None) as mock_method:
+        with patch.object(datapi.legacy_api_client.LegacyApiClient, "retrieve", return_value=None) as mock_method:
             get_land_sea_mask(
                 data_dir=str(data_dir), area=area, border=border
             )
@@ -159,7 +159,7 @@ class TestCDSDownloader(unittest.TestCase):
         output_filename = f"ERA5/monthly/era5_mean_sea_level_pressure_monthly_{start_year}-{end_year}.nc"
         output_path = Path(data_dir, output_filename)
 
-        with patch.object(cads_api_client.legacy_api_client.LegacyApiClient, "retrieve", return_value=None) as mock_method:
+        with patch.object(datapi.legacy_api_client.LegacyApiClient, "retrieve", return_value=None) as mock_method:
             get_era5_monthly(
                 data_dir=data_dir,
                 start_year=start_year,

